### PR TITLE
build(deps): Update dataflow dependencies

### PR DESCRIPTION
**Pull request by lc525**
**What this PR does / why we need it**:

Dataflow dependency updates (stage 1). Move to more recent Kotlin & Gradle releases, and update other dependencies as close to latest releases as is possible given compatibility constraints.

**Which issue(s) this PR fixes**:

Fixes # (internal Seldon issues)
- #INFRA-688 : klogging API updates
- #INFRA-687 : updates for working with newer grpc libraries

**Special notes for your reviewer**:
Some files are autogenerated by `gradle wrapper` on update to 8.5.0

### CODE REVIEW

File: scheduler/data-flow/gradlew
Comment (lc525) [inline scheduler/data-flow/gradlew:1]: Note for reviewer: file auto-generated by Gradle

File: scheduler/data-flow/gradle/wrapper/gradle-wrapper.properties
Comment (lc525) [inline scheduler/data-flow/gradle/wrapper/gradle-wrapper.properties:1]: Note for reviewer: file auto-generated by Gradle

File: scheduler/data-flow/gradlew.bat
Comment (lc525) [inline scheduler/data-flow/gradlew.bat:1]: Note for reviewer: file auto-generated by Gradle

**Review by sakoush — COMMENTED** [review]
Lgtm, maybe add a note somewhere with this issue with name resolution that you had to fix

**Review by sakoush — APPROVED** [review]

**Pull request merged**